src/modules/syncedVideoPlayer/SyncedVideoPlayer.ts

```diff
@@ -0,0 +1,78 @@
+import { Socket } from "socket.io-client";
+
+interface User {
+  id: string;
+  name: string;
+}
+
+class SyncedVideoPlayer {
+
+  /**
+   * User {id, name}
+   */
+  user: User
+
+  /**
+   * videoId - youtube videoId
+   */
+  videoId: String
+
+  /**
+   * channel - communication channel
+   */
+  channel: Socket
+
+  /**
+   * room - connected party socket room name
+   * Note: room and channel are the same thing in socket.io
+   */
+  room: String
+
+  /**
+   * player - Youtube player object
+   */
+  player: any
+
+  constructor(user: User, videoId: String, channel: Socket, room: String){
+    this.user = user;
+    this.videoId = videoId;
+    this.room = room;
+    this.channel = channel;
+
+    const id = 'some_id';
+    this.player = new window.YT.Player(`player-${id}`, {
+      videoId: this.videoId,
+      events: {
+        onReady: this.onPlayerReady,
+      },
+    });
+  }
+
+  onPlayerReady() {
+    // TODO
+  }
+  
+  /**
+   * start() is used to start the sync process, 
+   * once the video is selected on host side, all other connected users will be notified.
+   * server-side will send the notifiction to other clients in the room (messsaging to rooms is a server side feature)
```

Comment:
Where are we sending the the info to keep the videos in sync?

I was expecting for the same to happen inside `start`.

Reply:
That's what I had done earlier by sharing the startSeconds. But I am not sure now how to pass it to this function. Any suggestions on how do we get startSeconds here?

Also, I am reimplementing this function. I need some more time.

Reply:
Eventually the startSeconds value has to come from the channel. You've access to the channel here, use it.

Reply:
Somewhere it needs to be put in the channel too, would the `start()` method be the right place?
And how will I get it from the player without passing it as an argument every time?